exec/java-exec/src/main/java/org/apache/drill/exec/store/parquet/ParquetRecordWriter.java

```diff
@@ -302,6 +308,40 @@ private Type getType(MaterializedField field) {
     }
   }
 
+  /**
+   * Adds element type to {@code listBuilder} based on Drill's
+   * {@code elementField}.
+   *
+   * @param listBuilder list schema builder
+   * @param elementField Drill's type of list elements
+   */
+  private void addElementType(ListBuilder<GroupType> listBuilder, MaterializedField elementField) {
+    if (elementField.getDataMode() == DataMode.REPEATED) {
+      MaterializedField child2 = elementField.getChildren().iterator().next();
+      ListBuilder<GroupType> inner = org.apache.parquet.schema.Types.requiredList();
+      addElementType(inner, child2);
+      listBuilder.setElementType(inner.named("element"));
+    } else {
+      Type element = getType(elementField);
+      // element may have internal name '$data$',
+      // rename it to 'element' according to Parquet list schema
+      if (element.isPrimitive()) {
```

Comment:
Is it possible when the `element` is not primitive? Or if we currently don't assume such cases, should we throw an exception for non-primitive elements?

Reply:
yes, it's possible. Normally,  it's necessary to add another tests to check list of struct and repeated list of struct elements.